app/scripts/controllers/backup.js

```diff
@@ -0,0 +1,57 @@
+import { exportAsFile, prependZero } from '../../../ui/helpers/utils/util';
+
+export default class BackupController {
+  constructor(opts = {}) {
+    const {
+      preferencesController,
+      addressBookController,
+      trackMetaMetricsEvent,
+    } = opts;
+
+    this.preferencesController = preferencesController;
+    this.addressBookController = addressBookController;
+    this._trackMetaMetricsEvent = trackMetaMetricsEvent;
+  }
+
+  async restoreUserData(jsonString) {
+    console.log('jsonString: ', jsonString);
+    const { preferences, addressBook } = JSON.parse(jsonString);
+    preferences && this.preferencesController.store.updateState(preferences);
+    addressBook && this.addressBookController.update(addressBook, true);
+    this._trackMetaMetricsEvent({
+      event: 'User restored data from backup',
+      category: 'backup',
+    });
+  }
+
+  async backupUserData() {
+    const userData = {
+      preferences: this.preferencesController.store.getState(),
+      addressBook: this.addressBookController.state,
+    };
+
+    const result = JSON.stringify(userData);
+
+    const date = new Date();
+
+    const pz = (num) => prependZero(num, 2);
```

Comment:
I know it wouldn't make much difference, but can we update the constant name to `prefixZero` or something similar instead of `pz`.